llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -5277,7 +5277,7 @@ void SelectionDAGLegalize::PromoteNode(SDNode *Node) {
       Tmp1 = DAG.getNode(TruncOp, dl, Node->getValueType(0), Tmp1);
     else
       Tmp1 = DAG.getNode(TruncOp, dl, Node->getValueType(0), Tmp1,
-                         DAG.getIntPtrConstant(0, dl));
+                         DAG.getIntPtrConstant(0, dl, /*isTarget=*/true));
```

Comment:
We should stop using getIntPtrConstant and just make it a fixed i32 (then you could just use getTargetConstant everywhere)

Reply:
Could it be MVT::i1?

Reply:
Or that, yes. Really this should be an "exact" flag on fptrunc, but for some reason only SelectionDAG has this custom "fast math flag"